java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -860,18 +878,24 @@ class DatabaseImpl(
                     LEFT JOIN collection_entries
                         ON fields.is_collection = 1
                         AND collection_entries.collection_id = field_values.value_id
-                    WHERE fields.type_id = ?
+                    WHERE fields.type_id in (${typeIds.map { it.toString() }.joinToString()})
                 """.trimIndent()
 
             delete(
                 TABLE_NUMBER_PRIMITIVES,
-                "id NOT IN ($usedFieldIdsQuery)",
-                arrayOf(PrimitiveType.Number.ordinal.toString())
+                "id NOT IN (${usedFieldIdsQuery(
+                    listOf(
+                        PrimitiveType.Number.ordinal,
+                        PrimitiveType.Float.ordinal,
+                        PrimitiveType.Double.ordinal
+                    )
```

Comment:
Probably worth extracting this list out into a class-wide constant -- then you can use it up above in $numeric_types